### PR TITLE
Issue/core 323

### PR DESCRIPTION
**Pull request by LucasStarlingdePaulaSalles**
# Description

Kind: Task

Section: Authentication

Summary: refresh token function on auth service

Issue link: https://inspr.atlassian.net/browse/CORE-323

## Changelog

> Refresh endpoint

## Pay attention to

> Changes to tokenization

### CODE REVIEW

File: cmd/authsvc/api/controllers/refresh.go [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+// Refresh returns the refreshing endpoint. This entpoint receives a refresh token and a refresh url, it returns a refreshed token.
+func (server *Server) Refresh() rest.Handler {
+	return func(w http.ResponseWriter, r *http.Request) {
+		data := models.ResfreshDI{}
+		err := json.NewDecoder(r.Body).Decode(&data)
+		if err != nil {
+			err = ierrors.NewError().BadRequest().Message("invalid body").Build()
+			rest.ERROR(w, err)
+			return
+		}
```
Comment (ptcar2009) [inline cmd/authsvc/api/controllers/refresh.go:22]: we should really add a wrapper for encoding and decoding data, this gets old

File: cmd/authsvc/api/controllers/refresh.go [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+		payload := models.Payload{}
+		err = json.NewDecoder(resp.Body).Decode(&payload)
+		if err != nil {
+			rest.ERROR(w, err)
+			return
+		}
```
Comment (ptcar2009) [inline cmd/authsvc/api/controllers/refresh.go]: How do you feel about extracting this to a different function?

**Review by ptcar2009 — COMMENTED** [review]
LGTM otherwise

File: cmd/authsvc/api/controllers/refresh.go (hunk 2 above)
Comment (LucasStarlingdePaulaSalles) [reply cmd/authsvc/api/controllers/refresh.go]: I'll get right on it!

**Review by luiz-couto — APPROVED** [review]

**Review by ptcar2009 — APPROVED** [review]
LGTM

**Pull request merged**